pkg/tsdb/azuremonitor/macros_test.go

```
@@ -0,0 +1,155 @@
+			Err:       require.NoError,
+		},
+		{
+			name:      "$__interval should use the defined interval from the query",
+			timeRange: timeRange,
+			query: &tsdb.Query{
+				Model: simplejson.NewFromAny(map[string]interface{}{
+					"interval": "5m",
+				}),
+			},
+			kql:      "bin(TimeGenerated, $__interval)",
+			expected: "bin(TimeGenerated, 300000ms)",
+			Err:      require.NoError,
+		},
+		{
+			name:      "$__interval should use the default interval if none is specified",
+			timeRange: timeRange,
+			query: &tsdb.Query{
+				DataSource: &models.DataSource{},
+				Model:      simplejson.NewFromAny(map[string]interface{}{}),
+			},
+			kql:      "bin(TimeGenerated, $__interval)",
+			expected: "bin(TimeGenerated, 34000ms)",
+			Err:      require.NoError,
+		},
+		{
+			name:      "$__escapeMulti with multi template variable should replace values with KQL style escaped strings",
+			timeRange: timeRange,
+			query: &tsdb.Query{
+				DataSource: &models.DataSource{},
+				Model:      simplejson.NewFromAny(map[string]interface{}{}),
+			},
+			kql:      `CounterPath in ($__escapeMulti('\\grafana-vm\Network(eth0)\Total','\\grafana-vm\Network(eth1)\Total'))`,
+			expected: `CounterPath in (@'\\grafana-vm\Network(eth0)\Total', @'\\grafana-vm\Network(eth1)\Total')`,
+			Err:      require.NoError,
+		},
+		{
+			name:      "$__escapeMulti with multi template variable and has one selected value that contains comma",
+			timeRange: timeRange,
+			query: &tsdb.Query{
+				DataSource: &models.DataSource{},
+				Model:      simplejson.NewFromAny(map[string]interface{}{}),
+			},
+			kql:      `$__escapeMulti('\\grafana-vm,\Network(eth0)\Total Bytes Received')`,
+			expected: `@'\\grafana-vm,\Network(eth0)\Total Bytes Received'`,
+			Err:      require.NoError,
+		},
+		{
+			name:      "$__escapeMulti with multi template variable and is not wrapped in single quotes should fail",
```

should this return an error instead of silently failing

---

What is failing silently? `$__escapeMulti with multi template variable and is not wrapped in single quotes should fail` tests that it returns an error.

---

Argh my bad. I misread the `Err:      require.Error,` line.
I'm not used to checking for errors like that.